### PR TITLE
[extractor/youtube] Added is_post_live_dvr to returned info

### PR DESCRIPTION
**Pull request by lazypete365**
<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### Description of your *pull request* and other information

</details>

As I suggested in issue #1564, I have added `is_post_live_dvr` to the variables returned by Youtube downloads. Now, an expression like `--match-filter '!is_post_live_dvr'` should skip any videos that have not been fully processed by Youtube yet, and thus are potentially problematic. It is not a fix but I thought it might help as a workaround, in particular with archival scripts.

The value of the variable should be `is_post_live_dvr` if the video has not been fully processed (and thus could potentially encounter the issue if it is also >4h long), and `None` once it is.

EDIT: I forgot to mention that the variable name is a bit of a mouthful, and quite obscure. I only switched the original name within the extractor from camelCase to snake_case, to follow the naming convention of the other variables and avoid issues with the regex in the _match_one function. But if any maintainer comes up with a more intuitive and user-friendly variable name (or any better idea, in general) they are welcome to change it.

### CODE REVIEW

**Comment by pukkandan** [comment]
I understand the motivation, but extractors should not return arbitrary fields. It also doesn't make much sense to define a new field for this imo. What if we provided this info in `format_note` of all formats? or maybe set all formats to a lower `preference`? Either of these can be used in match-filter too

**Comment by pukkandan** [comment]
Or, better, se could set `live_status='post_live_dvr'`

**Comment by lazypete365** [comment]
I was about to say "maybe live_status could work too" :) . So I guess live_status it is.

**Comment by pukkandan** [comment]
POC
```diff
diff --git a/yt_dlp/extractor/common.py b/yt_dlp/extractor/common.py
index fc087a69c..c41225220 100644
--- a/yt_dlp/extractor/common.py
+++ b/yt_dlp/extractor/common.py
@@ -312,7 +312,8 @@ class InfoExtractor:
                     live stream that goes on instead of a fixed-length video.
     was_live:       True, False, or None (=unknown). Whether this video was
                     originally a live stream.
-    live_status:    'is_live', 'is_upcoming', 'was_live', 'not_live' or None (=unknown)
+    live_status:    None (=unknown), 'is_live', 'is_upcoming', 'was_live', 'not_live',
+                    or 'post_live' (was live, but VOD is not yet processed)
                     If absent, automatically set from is_live, was_live
     start_time:     Time in seconds where the reproduction should start, as
                     specified in the URL.
diff --git a/yt_dlp/extractor/youtube.py b/yt_dlp/extractor/youtube.py
index 02305c3f9..a5c4e00a6 100644
--- a/yt_dlp/extractor/youtube.py
+++ b/yt_dlp/extractor/youtube.py
@@ -3463,13 +3463,6 @@ def feed_entry(name):
             or get_first(microformats, 'lengthSeconds')
             or parse_duration(search_meta('duration'))) or None

-        if get_first(video_details, 'isPostLiveDvr'):
-            self.write_debug('Video is in Post-Live Manifestless mode')
-            if (duration or 0) > 4 * 3600:
-                self.report_warning(
-                    'The livestream has not finished processing. Only 4 hours of the video can be currently downloaded. '
-                    'This is a known issue and patches are welcome')
-
         live_broadcast_details, is_live, streaming_data, formats, automatic_captions = \
             self._list_formats(video_id, microformats, video_details, player_responses, player_url)

@@ -3600,6 +3593,14 @@ def feed_entry(name):
             'release_timestamp': live_start_time,
         }

+        if get_first(video_details, 'isPostLiveDvr'):
+            self.write_debug('Video is in Post-Live Manifestless mode')
+            if not is_live and (duration or 0) > 4 * 3600:
+                info['live_status'] == 'post_live'
+                self.report_warning(
+                    'The livestream has not finished processing. Only 4 hours of the video can be currently downloaded. '
+                    'This is a known issue and patches are welcome')
+
         subtitles = {}
         pctr = traverse_obj(player_responses, (..., 'captions', 'playerCaptionsTracklistRenderer'), expected_type=dict)
         if pctr:
```

**Comment by pukkandan** [comment]
PS: It needs to be documented in README too

**Comment by lazypete365** [comment]
If I understand it correctly, even if the video is shorter than 4 hours and does not encounter this particular issue, it is still going to be missing a bunch of formats until it has finished post-processing (at least that's what seemed to happen to the videos I've been using for testing). 

If that's the case, I think I would personally find the `post_live` status more useful if it was independet of the duration check for the warning, and just literally meant "post-processing live video". The duration check can always be added to the match-filter expression.

**Comment by pukkandan** [comment]
> If I understand it correctly, even if the video is shorter than 4 hours and does not encounter this particular issue, it is still going to be missing a bunch of formats until it has finished post-processing

You could just restrict the format codes for that

> If that's the case, I think I would personally find the post_live status more useful if it was independet of the duration check

that said, I don't mind this either. and technically, this is more correct

**Comment by pukkandan** [comment]
You forgot this
```diff
--- a/yt_dlp/extractor/common.py
+++ b/yt_dlp/extractor/common.py
@@ -312,7 +312,8 @@ class InfoExtractor:
                     live stream that goes on instead of a fixed-length video.
     was_live:       True, False, or None (=unknown). Whether this video was
                     originally a live stream.
-    live_status:    'is_live', 'is_upcoming', 'was_live', 'not_live' or None (=unknown)
+    live_status:    None (=unknown), 'is_live', 'is_upcoming', 'was_live', 'not_live',
+                    or 'post_live' (was live, but VOD is not yet processed)
                     If absent, automatically set from is_live, was_live
     start_time:     Time in seconds where the reproduction should start, as
                     specified in the URL.
```
Add similar explanation to readme too since it's not obvious what `post_live` means

**Comment by lazypete365** [comment]
👍

File: yt_dlp/extractor/youtube.py [hunk 1]
```diff
@@ -3599,6 +3592,15 @@ def feed_entry(name):
             'live_status': 'is_upcoming' if is_upcoming else None,  # rest will be set by YoutubeDL
             'release_timestamp': live_start_time,
         }
+        
+
```
Comment (pukkandan) [inline yt_dlp/extractor/youtube.py:3596]:
```suggestion

```

File: README.md [hunk 2]
```diff
@@ -1257,7 +1257,7 @@ The available fields are:
  - `average_rating` (numeric): Average rating give by users, the scale used depends on the webpage
  - `comment_count` (numeric): Number of comments on the video (For some extractors, comments are only downloaded at the end, and so this field cannot be used)
  - `age_limit` (numeric): Age restriction for the video (years)
- - `live_status` (string): One of "is_live", "was_live", "is_upcoming", "not_live"
+ - `live_status` (string): One of "is_live", "post_live" (was live, but VOD is not yet processed), "was_live", "is_upcoming", "not_live"
```
Comment (pukkandan) [inline README.md:1260]:
```suggestion
 - `live_status` (string): One of "not_live", "is_live", "is_upcoming", "was_live", "post_live" (was live, but VOD is not yet processed)
```

File: yt_dlp/extractor/youtube.py [hunk 3]
```diff
@@ -3599,6 +3592,13 @@ def feed_entry(name):
             'live_status': 'is_upcoming' if is_upcoming else None,  # rest will be set by YoutubeDL
             'release_timestamp': live_start_time,
         }
+        if get_first(video_details, 'isPostLiveDvr'):
```
Comment (pukkandan) [inline yt_dlp/extractor/youtube.py:3595]:
```suggestion

        if get_first(video_details, 'isPostLiveDvr'):
```

File: yt_dlp/extractor/common.py [hunk 4]
```diff
@@ -312,7 +312,8 @@ class InfoExtractor:
                     live stream that goes on instead of a fixed-length video.
     was_live:       True, False, or None (=unknown). Whether this video was
                     originally a live stream.
-    live_status:    'is_live', 'is_upcoming', 'was_live', 'not_live' or None (=unknown)
+    live_status:    None (=unknown), 'is_live', 'is_upcoming', 'was_live', 'not_live'
+                    or 'post_live' (was live, but VOD is not yet processed)
```
Comment (pukkandan) [inline yt_dlp/extractor/common.py:316]:
```suggestion
                    or 'post_live' (was live, but VOD is not yet processed)
```

**Comment by pukkandan** [comment]
Next time, make sure you run the linter

**Pull request merged**